assets/js/modules/analytics-4/utils/withCustomDimensions.js

```diff
@@ -270,29 +308,31 @@ export default function withCustomDimensions( options = {} ) {
 			useEffect( () => {
 				if (
 					! customDimensions ||
-					! isInvalidCustomDimensionError( reportError ) ||
+					! hasInvalidCustomDimensionError ||
 					isSyncingAvailableCustomDimensions
 				) {
 					return;
 				}
 
-				// Clear report error so that the fetch sync action isn't
-				// triggered multiple times.
-				clearError( 'getReport', [ reportOptions ] );
+				scheduleSyncAvailableCustomDimensions();
 
-				// Sync available custom dimensions.
-				fetchSyncAvailableCustomDimensions().then( () => {
-					// Invalidate report request so that it is re-fetched.
-					invalidateResolution( 'getReport', [ reportOptions ] );
+				// Clear report errors so that the useEffect isn't
+				// triggered multiple times.
+				Promise.all(
+					invalidCustomDimensionReportOptions.map( ( args ) => {
+						return clearError( 'getReport', [ args ] );
```

Comment:
The `return` here is unnecessary, correct?

Reply:
Even though the `return` is not necessary here, ESLint complains about the arrow function not having a return statement. We're also using a similar pattern in `create-snapshot-store.js`, e.g.

https://github.com/google/site-kit-wp/blob/14d70dea611e7de142ce3f6c122f9be15bbcae5a/assets/js/googlesitekit/data/create-snapshot-store.js#L187-L214

Thoughts?